.circleci/config.yml

```
@@ -294,3 +271,15 @@ commands:
             "$CIRCLE_PROJECT_USERNAME" \
             "$CIRCLE_PROJECT_REPONAME" \
             "$CIRCLE_BUILD_URL" > version.json
+
+  skip-if-do-not-deploy:
+    steps:
+      - run:
+          name: Check if deployment is disallowed
+          # This relies on the [do not deploy] text to be available in the
+          # merge commit when merging the PR to 'main'.
+          command: |
+            if git log -1 "$CIRCLE_SHA1" | grep -q '\[do not deploy\]'; then
+                echo "Skipping this step: deployment was disabled in the PR."
```

From what I know `circleci-agent step halt` will **not** terminate the current step. It will also **not** prevent other jobs in the workflow from running.

```suggestion
                echo "Skipping remaining steps in this job: deployment was disabled for this commit."
```